docs/pages/guides/nextjs-starter-kit.mdx

```diff
@@ -0,0 +1,545 @@
+    });
+    // ...
+  }
+  return <button onClick={handleCreateDocument}>New document</button>;
+}
+```
+
+Functions that return data can be used with [SWR hooks](https://swr.vercel.app/)
+that automatically update your data in components. For example,
+`getDocumentUsers` returns a list of users with access to the room:
+
+```tsx
+// Convert from this
+const { data, error } = await getDocumentUsers({
+  documentId: "my-document-id",
+});
+
+// To this
+const { data, error } = useDocumentsFunctionSWR([
+  getDocumentUsers,
+  {
+    documentId: "my-document-id",
+  },
+]);
+```
+
+Here's a working example:
+
+```tsx
+import { getDocumentUsers, useDocumentsFunctionSWR } from "../../lib/client";
+
+export function ListUsers() {
+  // Get users attached to a document and update every 1000ms
+  const { data: users, error: usersError } = useDocumentsFunctionSWR(
+    [
+      getDocumentUsers,
+      {
+        documentId: "my-document-id",
+      },
+    ],
+    { refreshInterval: 1000 }
+  );
+
+  if (usersError) {
+    return <div>Error</div>;
+  }
+
+  if (!users) {
+    return <div>Loading...</div>;
+  }
+
+  return (
+    <ul>
+      {users.map((user) => (
+        <li>user.name</li>
+      ))}
+    </ul>
+  );
+}
+```
+
+### How to extend the `Document` type
+
+If you'd like to add a new property to `Document`, it's simple. First, edit the
+`Document` type in `/types/documents.ts`:
+
+```ts file="types/documents.ts"
+export type Document = {
+  // Your new property
+  randomNumber: number;
+  //...
+};
+```
+
+Then modify the return value in `/lib/server/utils/buildDocuments.ts`. This is a
+function that converts a Liveblocks room into your custom document format:
+
+```ts file="lib/server/utils/buildDocuments.ts"
+// Return our custom Document format
+const document: Document = {
+  randomNumber: Math.random(),
+  // ...
+};
+```
+
+Next, run the following command to check for problems:
+
+```
+npm run typecheck
+```
+
+If no errors are returned, the document properties were successfully extended.
+
+### How to extend the `User` & `Session` type
+
+Similar to the way we extend the `Document` type, we can also extend the `User`
+and `Session` type.
+
+Adding a new property to `User`/`Session` is simple. First, edit the `User` type
+in `/types/data.ts`.
+
+```ts file="types/data.ts"
+export type User = {
+  // Your new property
+  randomNumber: number;
+  // ...
+};
+```
+
+Then make sure to return this new property in `/lib/server/database/getUser.ts`.
+
+```ts file="lib/server/database/getUser.ts"
+return { randomNumber: Math.random() /* ... */ };
+```
+
+The new property will now be available to use in your app:
+
+```ts
+// randomNumber: Math.random(),
+console.log(session.user.info.randomNumber);
+```
+
+#### Adding this to your Liveblocks app (optional)
+
+If you'd like your new property to be available inside your Liveblocks app (e.g.
+the React hooks used in whiteboard), you must modify `UserInfo` in
+`/liveblocks.config.ts`
```

Comment:
That’d be valuable to educate our users on use cases where this could be valuable. Not everyone would understand right away why they could need to pass property to a liveblocks app. 
We could give an example:
- If you need to have this info available when showing presence of your users
- or if you....